about/process/branch-naming.rst

```diff
@@ -6,57 +6,63 @@ Our branch-naming convention ensures software can be built by our CI and tested
++-------------------------------+-------------------------------------+------------------------+--------------------+
+| ``ubports/focal_-_<ext>``     | Branches which use the deprecated   | ``focal_-_<ext>``      | Ubuntu 20.04 LTS.  |
+|                               | "branch extensions" system for      |                        |                    |
+|                               | cross-components changes. No longer |                        |                    |
+|                               | recommended for usage.              |                        |                    |
++-------------------------------+-------------------------------------+------------------------+--------------------+
+| ``personal/<user>/<name>``,   | All other branches may be used for  | N/A, unless proposed as an MR against       |
+| and any other branches.       | proposing merge requests or for     | aforementioned branches (see below).        |
+|                               | other reasons. Theses branches are  |                                             |
+|                               | not built by the CI unless proposed |                                             |
+|                               | as a merge request.                 |                                             |
++-------------------------------+-------------------------------------+---------------------------------------------+
+
+To ensure the ``main`` branch doesn't miss any changes, all changes must be made against ``main`` branch first before backporting to the major release branches. An exception is when a change is not applicable to the ``main`` branch anymore due to other changes in that branch.
 
 .. note::
+    If the next development version is not available for testing yet, and you need the change to be made available for install for testing, it's allowed to temporarily target the changeset to the major version branch to get packages built against that version. When the changeset is ready for merge, the changeset must be re-targeted to the ``main`` branch.
 
-    There is no prioritization, so the build system will always use the package with the highest version number, or the newest build if the version is equal.
-
-Dependency-file
-^^^^^^^^^^^^^^^
+APT archives for Merge Requests
+-------------------------------
 
-For complex or non-linear dependencies, a ``ubports.depends`` file can be created in the root of the repository to specify additional dependencies. The branch name will be ignored if this file exists.
+In addition to archives mentioned above, we also create APT archives for MRs made against our Git repositories. This allows us to test the result of the build on real device using ``ubports-qa`` before merging the code. For each MR, the CI will build the code against all the same targets as the MR's target branch, and then will append ``_-_PR_<repository name>_<MR number>`` to them [1]_. For example:
```

Comment:
either "test the result of the build on *a* real device" or "test the result of the build on real *devices*"

Reply:
could we add a link here for ubports-qa ? where do people find documentation about this or the tool itself

Reply:
> either "test the result of the build on a real device" or "test the result of the build on real devices"

Fixed.

> could we add a link here for ubports-qa ? where do people find documentation about this or the tool itself

I linked the repo for ubports-qa itself, which has a readme documenting the tool. That said, seems like I'll have to update that tool too for the new MR archive format.